pkg/collector/metric/types/types.go

```diff
@@ -85,28 +78,40 @@ type UInt64StatCollection struct {
 }
 
 func (s *UInt64StatCollection) SetAggrStat(key string, newAggr uint64) {
-	if _, found := s.Stat[key]; !found {
-		s.Stat[key] = &UInt64Stat{}
-	}
-	if err := s.Stat[key].SetNewAggr(newAggr); err != nil {
-		klog.V(3).Infoln(err)
+	if instance, found := s.Stat[key]; !found {
+		s.Stat[key] = &UInt64Stat{
+			Aggr:  newAggr,
+			Delta: newAggr,
```

Comment:
The first Delta value should be 0

Reply:
but in my point of view, as the UInt64Stat default is Aggr:0, Delta:0 when SetNewAggr been invoked with a no zero value x, it turn to  Aggr:x, Delta:x right?

Reply:
hence, I suppose for any new SetAggrStat/AddDeltaStat/SetDeltaStat with a new record, it can be init as 
```
&UInt64Stat{
			Aggr:  newAggr,
			Delta: newAggr,
}
```
as from 0 to newAggr?

Reply:
There is a slight difference in the logic between adding aggregated values (i.e., counter) and gauge values (i.e., delta), mainly for the first added value.

When adding an aggregated value (`SetAggrStat`), the delta value represents the difference between the new aggregated value and the previous aggregated value. Therefore, if no previous aggregated value exists, the delta should be zero.

In contrast, when adding a delta value (`AddDeltaStat`/`SetDeltaStat`), which is a metric that is not a counter but a gauge, the first value of both the Aggr and Delta will be the same. Since the Aggr is the sum of all added Deltas.

Btw, it may be worthwhile to consider renaming the values using the Prometheus nomenclature, using `counter` for aggregated values and `gauge` for delta values.

Reply:
ok, got it.